contentcuration/contentcuration/utils/recommendations.py

```diff
@@ -61,67 +68,200 @@ def create_backend(self) -> Backend:
+        :return: The recommendations for the given topic. :rtype: RecommendationsResponse
+        """
+        recommendations = []
+        request = EmbedTopicsRequest(
+            params={'override_threshold': override_threshold},
+            json=topic,
+        )
+        response = self.generate_embeddings(request=request)
+        nodes = self._extract_data(response)
+        if len(nodes) > 0:
+            node_ids = [node['contentnode_id'] for node in nodes]
+            recommendations = list(ContentNode.objects.filter(id__in=node_ids))
+
+        return RecommendationsResponse(results=recommendations)
+
+    def _extract_data(self, response: BackendResponse) -> List[Dict[str, Any]]:
+        return response.data if response.data is not None else []
+
+    def embed_content(self, nodes: List[ContentNode]) -> bool:
+        """
+        Embeds the content for the given nodes. This is an asynchronous process and could take a
+        while to complete. This process is handled by our curriculum automation service.
+        See https://github.com/learningequality/curriculum-automation
+
+        :param nodes: The nodes for which to embed the content.
+        :return: A boolean indicating that content embedding process has started.
+        :rtype: bool
+        """
+        for i in range(0, len(nodes), 20):
+            batch = nodes[i:i + 20]
+            content = [self.extract_content(node) for node in batch]
+            request = EmbedContentRequest(json=content)
+            self.generate_embeddings(request=request, cache=False)
+
+        return True
 
     def extract_content(self, node: ContentNode) -> Dict[str, Any]:
-        return {}
+        """
+        Extracts the content from the given node.
+
+        :param node: The node from which to extract the content.
+        :return: A dictionary containing the extracted content.
+        :rtype: Dict[str, Any]
+        """
+        contentkind_to_presets = {
+            content_kinds.AUDIO: [format_presets.AUDIO, format_presets.AUDIO_DEPENDENCY],
+            content_kinds.VIDEO: [
+                format_presets.VIDEO_DEPENDENCY,
+                format_presets.VIDEO_HIGH_RES,
+                format_presets.VIDEO_LOW_RES,
+                format_presets.VIDEO_SUBTITLE,
+            ],
+            content_kinds.EXERCISE: [format_presets.DOCUMENT, format_presets.EPUB],
+            content_kinds.DOCUMENT: [format_presets.DOCUMENT, format_presets.EPUB],
+            content_kinds.HTML5: [format_presets.HTML5_ZIP],
+            content_kinds.H5P: [format_presets.H5P_ZIP],
+            content_kinds.ZIM: [format_presets.ZIM],
+        }
+
+        contentkind = node.kind
+        presets = contentkind_to_presets.get(contentkind.kind)
+        files = self._get_content_files(node, presets) if presets else None
+
+        try:
+            channel = Channel.object.filter(main_tree=node).first()
```

Comment:
This would not fetch the channel for the node. I believe I gave feedback or we talked about having channel ID passed in along with the node, since in most contexts where we're generating embeddings (and extracting files) we would have that context.